web-frontend/src/custom/api/usePost.ts

```diff
@@ -1,22 +1,42 @@
-// STYLES
-
-// EXTERNALS
 import { UseMutationResult, useMutation } from '@tanstack/react-query';
-import axios from 'axios';
-
-// LOCALS
+import axios, { AxiosResponse } from 'axios';
 
-export function usePost<T>(
+/**
+ * @param resource - path of the resource
+ * @typeParam TBody - type of the request
+ * @typeParam TResponse - type of the response
+ * @returns
+ */
+export function usePost<TBody, TResponse = null>(
```

Comment:
is this the same as what we did in station ?

Reply:
copy paste